src/ape/contracts/base.py

```diff
@@ -159,6 +159,8 @@ def __repr__(self) -> str:
         return self.abi.signature
 
     def serialize_transaction(self, *args, **kwargs) -> TransactionAPI:
+        if "sender" in kwargs and isinstance(kwargs["sender"], ContractInstance):
```

Comment:
What does the failure look like when not using a provider that can impersonate?

Reply:
```
impersonating vitalik and sending 1 wei
Traceback (most recent call last):
  File "/home/doge/ape/ape/src/ape/managers/accounts.py", line 80, in __getitem_str
    can_impersonate = self.provider.unlock_account(account_id)
  File "/home/doge/ape/ape/src/ape/utils.py", line 768, in inner
    raise NotImplementedError(
NotImplementedError: Attempted to call method 'ProviderAPI.unlock_account', method not supported.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/doge/.pyenv/versions/apeworx/bin/ape", line 33, in <module>
    sys.exit(load_entry_point('eth-ape', 'console_scripts', 'ape')())
  File "/home/doge/.pyenv/versions/apeworx/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/doge/.pyenv/versions/apeworx/lib/python3.9/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/doge/ape/ape/src/ape/_cli.py", line 40, in invoke
    return super().invoke(ctx)
  File "/home/doge/.pyenv/versions/apeworx/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/doge/ape/ape/src/ape_run/_cli.py", line 23, in invoke
    return super().invoke(ctx)
  File "/home/doge/.pyenv/versions/apeworx/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/doge/ape/ape/src/ape/cli/commands.py", line 18, in invoke
    super().invoke(ctx)
  File "/home/doge/.pyenv/versions/apeworx/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/doge/.pyenv/versions/apeworx/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/doge/ape/ape/src/ape_run/_cli.py", line 86, in call
    ns = run_module(f"scripts.{filepath.stem}")
  File "/home/doge/.pyenv/versions/3.9.9/lib/python3.9/runpy.py", line 213, in run_module
    return _run_code(code, {}, init_globals, run_name, mod_spec)
  File "/home/doge/.pyenv/versions/3.9.9/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/doge/ape/dummyproject/scripts/test.py", line 8, in <module>
    vitalik = accounts["0xab5801a7d398351b8be11c439e05c5b3259aec9b"]
  File "/home/doge/.pyenv/versions/3.9.9/lib/python3.9/functools.py", line 927, in _method
    return method.__get__(obj, cls)(*args, **kwargs)
  File "/home/doge/ape/ape/src/ape/managers/accounts.py", line 296, in __getitem_str
    return self.test_accounts[account_id]
  File "/home/doge/.pyenv/versions/3.9.9/lib/python3.9/functools.py", line 927, in _method
    return method.__get__(obj, cls)(*args, **kwargs)
  File "/home/doge/ape/ape/src/ape/managers/accounts.py", line 83, in __getitem_str
    raise IndexError(
IndexError: Your provider does not support impersonating accounts:
No account with address '0xAb5801a7D398351b8bE11C439e05C5B3259aeC9B'.
```

Reply:
> Are you using hardhat in mainnet-fork?

We were specifically checking to see what it looks like when not using a provider like `Hardhat` mainnet-fork.
I think the error is expected actually... It is an `IndexError` with some ape-flair. It is not an `ApeException`, that is why it has a lot of trace... But I think that is good in this situation.

Reply:
No error when using hardhat mainnet-fork. Above error is when using default provider. Error looks good and is expected.